owncloudComLibrary/src/main/java/com/owncloud/android/lib/common/http/methods/HttpBaseMethod.kt

```diff
@@ -13,24 +13,39 @@ import java.net.MalformedURLException
 import java.net.URL
 import java.util.concurrent.TimeUnit
 
-abstract class HttpBaseMethod constructor(clientWrapper: HttpClient, url: URL) {
-    var okHttpClient: OkHttpClient
+abstract class HttpBaseMethod constructor(url: URL) {
     var httpUrl: HttpUrl = url.toHttpUrlOrNull() ?: throw MalformedURLException()
     var request: Request
     abstract var response: Response
-
     var call: Call? = null
 
+    var followRedirects: Boolean? = true
+    var retryOnConnectionFailure: Boolean? = false
+    var connectionTimeoutVal: Long? = null
+    var connectionTimeoutUnit: TimeUnit? = null
+    var readTimeoutVal: Long? = null
+    var readTimeoutUnit: TimeUnit? = null
```

Comment:
Just a suggestion that could be out of scope from this PR. 

We could create a new data class that receives the value and the `TimeUnit` and use it here. So we could replace 2 variables with just one, and we won't need nested `?.let` to set the timeouts. Maybe a new issue to do this?